### DISCUSSION TITLE
Reloading the displayed dynamic page gives an error

### DISCUSSION THREAD

**Discussion by AlexDernov** [neuefische/questions#662] [Web] [answered]
Hello. 

I have a problem with the stability of details pages (pages/locations/[id].js). 

The transition to the details page from locations page (pages/locations  - list of locations) goes without problems, but if I manually enter the final route (with id of any location from the list) or reload the displayed details page - it shows me an error ("failed to load"). 

This does not happen with API routes (api/locations/[id].js)! 
My data is in MongoDB.

Maybe someone already had a similar problem? I'm fighting with this for the third day (google, rewriting code including fetch functions, etc.).

My current PR: https://github.com/AlexDernov/capestone-project-hungryOar/pull/10

Thank you in advance!

**Comment by ThomasF85** [comment, marked as answer]
In your _app.js you have the following code:
```
const { data, isLoading, error } = useSWR("api/locations", fetcher);

  if (isLoading) {
    return <h1>Loading...</h1>;
  }
  if (error) return <div>failed to load</div>;
```

This is where the "failed to load" comes from.

Try correcting your SWR call to:
`const { data, isLoading, error } = useSWR("/api/locations", fetcher);`

**Reply by AlexDernov** [reply]
My goodness! Sure! Only I can be so blind! Thank you very much!
